course_catalog/tasks.py

```diff
@@ -0,0 +1,129 @@
+"""
+course_catalog tasks
+"""
+import logging
+
+import requests
+import boto3
+from ocw_data_parser import OCWParser
+from celery.task import task
+from django.conf import settings
+from course_catalog.constants import NON_COURSE_DIRECTORIES
+from course_catalog.models import Course
+from course_catalog.tasks_helpers import (
+    get_access_token,
+    parse_mitx_json_data,
+    safe_load_json,
+    digest_ocw_course,
+    get_s3_object_and_read,
+)
+
+
+log = logging.getLogger(__name__)
+
+
+@task
+def get_edx_data(force_overwrite=False):
+    """
+    Task to sync mitx data with the database
+    Args:
+        force_overwrite (bool): A boolean value to force the incoming course data to overwrite existing data
+    """
+    url = settings.EDX_API_URL
+
+    while url:
+        access_token = get_access_token()
+        response = requests.get(url, headers={"Authorization": "JWT " + access_token})
+        if response.status_code == 200:
+            for course_data in response.json()["results"]:
+                try:
+                    parse_mitx_json_data(course_data, force_overwrite)
+                except Exception:  # pylint: disable=broad-except
+                    log.exception("Error encountered parsing MITx json")
+        else:
+            log.error("Bad response status %s for %s", str(response.status_code), url)
+            break
+
+        url = response.json()["next"]
+
+
+@task
+def get_ocw_data():
+    """
+    Task to sync OCW course data with database
+    """
+    raw_data_bucket = boto3.resource(
```

Comment:
replaced the list with a dict

Reply:
Reverting this change, since it increased memory usage by over 4x:

<img width="1185" alt="screen shot 2019-02-06 at 8 01 39 am" src="https://user-images.githubusercontent.com/187676/52344809-b8705880-29e9-11e9-886a-ebf1d964c226.png">